app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -585,6 +615,22 @@ private void setSort(DBHelper.LoyaltyCardOrder order, DBHelper.LoyaltyCardOrderD
         updateLoyaltyCardList(false);
     }
 
+    private void setArchivedSort(DBHelper.LoyaltyCardArchiveFilter filter){
+        // Update values
+        mArchiveFilter = filter;
+
+        // Store in Shared Preference to restore next app launch
+        SharedPreferences sortPref = getApplicationContext().getSharedPreferences(
+                getString(R.string.sharedpreference_sort),
+                Context.MODE_PRIVATE);
+        SharedPreferences.Editor sortPrefEditor = sortPref.edit();
+        sortPrefEditor.putString(getString(R.string.sharedpreference_sort_filter), filter.name());
```

Comment:
I think this line should just be part of setSort, and this whole function is unnecessary

Reply:
And I'd rather not name this "filter", that's too generic. It's also not really related to sorting, we're just putting it in the same menu as it seems to fit nicely. I'd say `sharedpreference_show_archived` would be better.